### ISSUE TITLE
How to make Odata request binding case insensitive

### ISSUE THREAD

**Issue by brijeshb** [OData/WebApi#2247]
*How do i make odata bind request json in a case insensitive manner? I really appreciated this feature in web api and now as i try toward better odata compatibility this feels like an unnecessary hindrance on my clients. (I'm currently using the latest release on .net core. If there does exist documentation about this, please direct me to it.)*

**Comment by GreenEyedAndy** [comment]
Any progress here?
We have the same problem. All is working well GET, POST, DELETE but PATCH isn't working because of the different case sensitive handling.

**Comment by KenitoInc** [comment]
@GreenEyedAndy We have a PR under review. We hope to have the fix in the next release.

**Comment by jsgoupil** [comment]
I appreciate the work being done on this. This worked fine prior the version 7, but when moving to .NET Core, then case sensitivity started to bite us.
I am happy we are going towards a fix for this, but is there a way we can workaround before this gets in? Because I have doubts it will happen quickly.

I am interested in this `Delta` that is now case sensitive in .NET Core.

I spent about 4h digging in plenty of code trying to figure out the right way to go around this.

**Comment by jsgoupil** [comment]
Alright, another 2h spent. I have noticed a couple of things:

- The `ODataConventionModelBuilder` has something like `EnableLowerCamelCase()`, and when enabled, it just works!
- However, I don't use EdmModel like this in .NET Core. I actually just used `Delta<T>`, and I have noticed that it's using the base of the default Json Serializer installed with .NET Core. I have Newtonsoft, configured to use the naming strategy of CamelCase. However, the `Delta<T>` does not like receiving a `camelCase` for property.

After spending so much time on this, I only needed the .Patch method to work. I will simply drop the OData and come up with my own Pathcher. Something more manual like this:

https://stackoverflow.com/a/19662467/734742
